src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -0,0 +1,170 @@
+     * commands, globally mute other players.</p>
+     */
+    static final String RANK_STAFF = "staff";
+    /**
+     * A permission with this rank should only be granted to fully trusted
+     * server administrators. The permission's impact on the server may be high.
+     *
+     * <p>Examples can be create/delete worlds, grant permissions, restart
+     * server.</p>
+     */
+    static final String RANK_ADMIN = "admin";
+    /**
+     * A permission with this rank should normally not granted to anybody. This
+     * usually applies to debug permissions.
+     */
+    static final String RANK_NOBODY = "nobody";
+
+    /**
+     * Gets the permission id this description belongs too. May include
+     * placeholders {@code "<SomeParameter>"} such as in
+     * {@code "pluginId.give.item.<ItemType>"}.
+     *
+     * @return The permission id
+     */
+    String getId();
+
+    /**
+     * Gets a short description of the linked permission. May include a link to
+     * a more detailed description on the plugin's web page.
+     *
+     * @return A short description of the linked permission
+     */
+    Text getDescription();
+
+    /**
+     * Gets the human readable <b>suggested</b> rank the players should have
+     * before they should have access to this permission.
+     *
+     * <p><b>Note:</b> This is a suggestion for server owners and does not have
+     * any impact on the results of the permission checks.</p>
+     *
+     * @return The suggested player rank the permission should be granted to
+     */
+    Optional<String> getSuggestedRank();
+
+    /**
+     * Gets the owning plugin the permission belongs to.
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    PluginContainer getOwner();
+
+    /**
+     * A builder for permission descriptions.
+     */
+    interface Builder {
+
+        /**
+         * Sets the permission id for the description this builder creates. May
+         * include placeholders {@code "<SomeParameter>"} such as in
+         * {@code "pluginId.give.item.<ItemType>"}.
+         *
+         * @param permissionId The permission id
+         * @return This builder for chaining
+         */
+        Builder id(String permissionId);
+
+        /**
+         * Sets the short description to use. May include a link to a more
+         * detailed description on the plugin's web page.
+         *
+         * @param description The short description to use
+         * @return This builder for chaining
+         */
+        Builder description(Text description);
+
+        /**
+         * Sets the suggested rank players should have before being able to
+         * access this permission. Setting this is optional although
+         * recommended.
+         *
+         * @param rank The suggested rank to have
+         * @return This builder for chaining
+         */
+        Builder suggestedRank(@Nullable String rank);
```

Comment:
So the way I brought up of doing this was setting permissions on `rank-template` type subjects (or maybe `role`, just a name), which could then be added as parents to the server-specific groups. Adding this method just makes another thing for permissions plugins to handle when thinking about what permissions to assign where.